src/bio/smd/smd_pool.c

```diff
@@ -342,13 +342,25 @@ smd_pool_list_cb(struct sys_db *db, char *table, d_iov_t *key, void *args)
 	enum smd_dev_type	 st;
 	struct d_uuid            id;
 	int                      rc;
+	bool			 meta_pool = false;
+
+	if (!strncmp(table, TABLE_POOLS[SMD_DEV_TYPE_META], SMD_DEV_NAME_MAX))
+		meta_pool = true;
 
 	D_ASSERT(key->iov_len == sizeof(id));
 	id = *(struct d_uuid *)key->iov_buf;
 	memset(pools, 0, sizeof(struct smd_pool) * SMD_DEV_TYPE_MAX);
 	for (st = SMD_DEV_TYPE_DATA; st < SMD_DEV_TYPE_MAX; st++) {
 		rc = smd_db_fetch(TABLE_POOLS[st], &id, sizeof(id), &pools[st], sizeof(pools[st]));
```

Comment:
why do we fetch from all TABLE_POOLS, instead of just the current "table", since you have already tried both META and DATA in smd_pool_list()

Reply:
So for MD-on-SSD case, if we fetch succeed from META firstly, we won't try DATA.
For PMDK case, we don't have Meta, we will try to fetch something from data.

actually META or Data is acutally iterated, this is somehow for back-compatible reason.